src/main/scala/org/broadinstitute/dsde/workbench/sam/dataAccess/PostgresAccessPolicyDAO.scala

```diff
@@ -1723,28 +1726,34 @@ class PostgresAccessPolicyDAO(
             left join ${RoleActionTable as roleAction} on ${effectivePolicyRole.resourceRoleId} = ${roleAction.resourceRoleId}
             left join ${ResourceActionTable as resourceAction} on ${roleAction.resourceActionId} = ${resourceAction.id}
             left join ${ResourceTable as resource} on ${effectiveResourcePolicy.resourceId} = ${resource.id}
+            left join ${AuthDomainTable as authDomain} on ${authDomain.resourceId} = ${resource.id}
+            left join ${GroupTable as authDomainGroup} on ${authDomainGroup.id} = ${authDomain.groupId}
+            left join ${GroupMemberFlatTable as authDomainGroupMemberFlat} on ${authDomainGroup.id} = ${authDomainGroupMemberFlat.groupId} and ${authDomainGroupMemberFlat.memberUserId} = ${samUser.id}
```

Comment:
I assume this flat table is probably pretty big any worries about performance?

Reply:
Looking at the query plan, the flat table used for Auth Domains is being index-scanned, and the cost doesn't look like anything to worry about. Getting literally all resources I have access to in dev, totaling 3,199 rows including public resources, only takes 75ms